src/storage/mem_table_snapshot.cc

```diff
@@ -512,24 +543,64 @@ int MemTableSnapshot::MakeSnapshot(std::shared_ptr<Table> table, uint64_t& out_o
     return ret;
 }
 
-int MemTableSnapshot::RemoveDeletedKey(const ::openmldb::api::LogEntry& entry, const std::set<uint32_t>& deleted_index,
+int MemTableSnapshot::RemoveDeletedKey(std::shared_ptr<Table> table, const ::openmldb::api::LogEntry& entry,
+                                       const std::set<uint32_t>& deleted_index,
                                        std::string* buffer) {
     uint64_t cur_offset = entry.log_index();
     if (entry.dimensions_size() == 0) {
         std::string combined_key = entry.pk() + "|0";
+        if (entry.has_ts()) {
+            combined_key += "|" + std::to_string(entry.ts());
+        }
         auto iter = deleted_keys_.find(combined_key);
         if (iter != deleted_keys_.end() && cur_offset <= iter->second) {
             DEBUGLOG("delete key %s  offset %lu", entry.pk().c_str(), entry.log_index());
             return 1;
         }
     } else {
+        uint32_t ts_index = UINT32_MAX;
         std::set<int> deleted_pos_set;
         for (int pos = 0; pos < entry.dimensions_size(); pos++) {
+            bool use_ts = table->CheckTsInIndex(entry.dimensions(pos).idx(), &ts_index) ? false : true;
             std::string combined_key = entry.dimensions(pos).key() + "|" + std::to_string(entry.dimensions(pos).idx());
+            std::string combined_key_ts = "";
+            if (!use_ts) {
+                std::string buff;
+                openmldb::base::Slice data;
+                if (table->GetCompressType() == openmldb::type::kSnappy) {
+                    snappy::Uncompress(entry.value().data(), entry.value().size(), &buff);
+                    data.reset(buff.data(), buff.size());
+                } else {
+                    data.reset(entry.value().data(), entry.value().size());
+                }
+                const int8_t* raw = reinterpret_cast<const int8_t*>(data.data());
+                uint8_t version = openmldb::codec::RowView::GetSchemaVersion(raw);
+                std::vector<std::string> value_vec;
+                std::shared_ptr<Schema> schema = table->GetVersionSchema(version);
```

Comment:
decode ts column refer [here](https://github.com/4paradigm/OpenMLDB/blob/main/src/storage/mem_table.cc#L397)

Reply:
call `GetVersionDecoder` to get `RowView` rather than schema